leecade/react-native-swiper
# Fix invalid offset when loop option is used

### Is it a bugfix ?

- Yes
- fix #617, #632, #635

### Is it a new feature ?

- No

### Describe what you've done:

- take `loop` property into account when calculating new offset in `initState` function
- use `initState.width` and `initState.height` instead of `window.width` and `window.height` to calculate new offset in `initState` function
- new offset is also saved to `this.internals` in `initState` function
- refactor `initState` function a little bit

https://github.com/leecade/react-native-swiper/pull/646 claims to fix the same issues but fixes in that PR didn't help in my case.

### How to test it ?

- create swiper with several slides and `loop={true}`
- go to the 1st slide
- swipe back to go to the last slide
- swipe forth to go to the 1st slide again
- refresh page (I'm using `RefreshControl` component)

=>

- swiper should display the 1st slide - not the last one
- x offset of the slide should be calculated correctly on iOS

also I noticed (using `console.log(this.state.offset)` in `renderScrollView` function) that x offset is now correctly updated when swiping back and forth (it wasn't the case in 1.15.13 - sometimes x offset remained the same even after switching to another slide).

P.S. or else see linked issues for steps to reproduce the bug.
